### DISCUSSION TITLE
I hesitate to put this as a bug, however when I test the app locally and ask for pikachu info I get the following error.

### DISCUSSION THREAD

**Discussion by timstew** [boldare/openai-assistant#62] [Q&A]
/Users/tim/Projects/openai-assistant/node_modules/openai/src/error.ts:70
      return new BadRequestError(status, error, message, headers);
             ^
Error: 400 Runs in status "in_progress" do not accept tool outputs.
    at Function.generate (/Users/tim/Projects/openai-assistant/node_modules/openai/src/error.ts:70:14)
    at OpenAI.makeStatusError (/Users/tim/Projects/openai-assistant/node_modules/openai/src/core.ts:383:21)
    at OpenAI.makeRequest (/Users/tim/Projects/openai-assistant/node_modules/openai/src/core.ts:446:24)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at AssistantStream._createToolAssistantStream (/Users/tim/Projects/openai-assistant/node_modules/openai/src/lib/AssistantStream.ts:224:20)
    at AssistantStream._runToolAssistantStream (/Users/tim/Projects/openai-assistant/node_modules/openai/src/lib/AbstractAssistantStreamRunner.ts:246:12)

**Comment by timstew** [comment]
Other times (after figuring out how to turn the logging on), when clicking the suggested button to ask about pikachu stats, I can see in the logging that it is replying about being confused and needing more context, however this reply does not make it to the UI, and I never get a reply.

This is on unmodified code other than adding my own Api keys for openai and the weather service (which mostly works)

I'm trying to make sure i fully understand these agents before I decide to adapt/implement my own on this platform.

**Reply by sebastianmusial** [reply]
Hello @timstew 👋 
Thank you for mentioning the issue. Unfortunately, I can't reproduce it in my local environment.

Could you share with me the following information:
1. What Node version do you use?
2. What versions of OpenAI and @boldare/openai-assistant do you use?

**Reply by timstew** [reply]
I instantly felt mortified when I realized that I hadn't updated node in a while...  after upgrading to the latest LTS, v22.2.0

I do see a few deprecated warnings when starting up that weren't there before however the issue persists:

[Nest] 46026  - 2024-05-27, 5:09:54 p.m.     LOG [NestApplication] Nest application successfully started +1102ms
[Nest] 46026  - 2024-05-27, 5:09:54 p.m.     LOG 🚀 Application is running on: http://localhost:3000/api
[Nest] 46026  - 2024-05-27, 5:09:58 p.m.     LOG [ChatGateway] Client connected
[Nest] 46026  - 2024-05-27, 5:09:58 p.m.     LOG [ChatGateway] Client connected
[Nest] 46026  - 2024-05-27, 5:10:18 p.m.     LOG [ChatGateway] Socket "callStart" | threadId thread_Rt4qdioY18GaeXYlDkdNFzsd | files: 0 | content: Show me the stats for Pikachu (Pokémon)?
[Nest] 46026  - 2024-05-27, 5:10:21 p.m.     LOG [ChatGateway] Socket "runStepCreated": in_progress
[Nest] 46026  - 2024-05-27, 5:10:21 p.m.     LOG [ChatGateway] Socket "toolCallCreated": call_JtsCzKYQWoRXqCMeb3xxfjJH
[Nest] 46026  - 2024-05-27, 5:10:21 p.m.     LOG [ChatGateway] Socket "runStepDelta": in_progress
[Nest] 46026  - 2024-05-27, 5:10:21 p.m.     LOG [ChatGateway] Socket "toolCallDelta": call_JtsCzKYQWoRXqCMeb3xxfjJH
[Nest] 46026  - 2024-05-27, 5:10:21 p.m.     LOG [ChatGateway] Socket "runStepDelta": in_progress
[Nest] 46026  - 2024-05-27, 5:10:21 p.m.     LOG [ChatGateway] Socket "toolCallDelta": call_JtsCzKYQWoRXqCMeb3xxfjJH
[Nest] 46026  - 2024-05-27, 5:10:21 p.m.     LOG [ChatGateway] Socket "runStepDelta": in_progress
[Nest] 46026  - 2024-05-27, 5:10:21 p.m.     LOG [ChatGateway] Socket "toolCallDelta": call_JtsCzKYQWoRXqCMeb3xxfjJH
[Nest] 46026  - 2024-05-27, 5:10:21 p.m.     LOG [ChatGateway] Socket "runStepDelta": in_progress
[Nest] 46026  - 2024-05-27, 5:10:21 p.m.     LOG [ChatGateway] Socket "toolCallDelta": call_JtsCzKYQWoRXqCMeb3xxfjJH
[Nest] 46026  - 2024-05-27, 5:10:21 p.m.     LOG [ChatGateway] Socket "runStepDelta": in_progress
[Nest] 46026  - 2024-05-27, 5:10:21 p.m.     LOG [ChatGateway] Socket "toolCallDelta": call_JtsCzKYQWoRXqCMeb3xxfjJH
[Nest] 46026  - 2024-05-27, 5:10:21 p.m.     LOG [ChatGateway] Socket "runStepDelta": in_progress
[Nest] 46026  - 2024-05-27, 5:10:21 p.m.     LOG [ChatGateway] Socket "toolCallDelta": call_JtsCzKYQWoRXqCMeb3xxfjJH
[Nest] 46026  - 2024-05-27, 5:10:21 p.m.     LOG [ChatGateway] Socket "runStepDelta": in_progress
[Nest] 46026  - 2024-05-27, 5:10:22 p.m.     LOG [ChatGateway] Socket "toolCallDelta": call_JtsCzKYQWoRXqCMeb3xxfjJH
[Nest] 46026  - 2024-05-27, 5:10:22 p.m.     LOG [ChatGateway] Socket "runStepDelta": in_progress
[Nest] 46026  - 2024-05-27, 5:10:22 p.m.     LOG [ChatGateway] Socket "toolCallDone": call_JtsCzKYQWoRXqCMeb3xxfjJH
[Nest] 46026  - 2024-05-27, 5:10:24 p.m.     LOG [ChatGateway] Socket "runStepDone": completed
[Nest] 46026  - 2024-05-27, 5:10:36 p.m.     LOG [ChatGateway] Socket "callDone" | threadId thread_Rt4qdioY18GaeXYlDkdNFzsd | content: Seems I'm lost, would you mind reformulating your question

<img width="673" alt="Screen Shot 2024-05-27 at 5 11 17 PM" src="https://github.com/boldare/openai-assistant/assets/12635792/b6d3e215-3d22-4a24-9f3b-e1c216a99b6d">

It's not totally clear to me where to provide additional nudging with respect to tool use. The prompt I see in the assistant instructions doesn't mention any of the tools. Is all that context only coming from the FunctionTool descriptions in the agent code?

**Reply by timstew** [reply]
One other thing to note, I've recently changed the model to gpt-4o but that doesn't appear to have any impact.

**Reply by timstew** [reply]
I neglected to state what version of the code I'm running.
It was the latest code from this repo, as of yesterday. I noticed there was a new commit about 9 hours ago, so that wasn't in there.

So far I've written two agents, one of them works flawlessly, and the other one I'm unable to trigger the tool call through the chat, which is why I want to really understand this pokemon issue, although with my second agent, I'm not getting any message about being lost, it just never tries to call the tool and sends messages back about trying again later

**Reply by sebastianmusial** [reply]
@timstew Sorry for the delayed response. I've tried to reproduce your issue multiple times with various configurations, but I haven't encountered the same behavior.

We recently released a new version, so I hope the update will magically resolve your issue. 🪄

**Reply by olivierhalupczok** [reply]
@timstew If you encounter further problems you can use this link to book a call with us, so we can see the issue ourselves and help you: [book a call](https://calendly.com/olivier-halupczok/30min)

**Comment by rmcsharry** [comment]
@olivierhalupczok Your demo page seems to suffer the same problem.

![Screenshot 2024-08-19 at 16 14 55](https://github.com/user-attachments/assets/20cd522e-9ec0-477c-98e2-57fa28a6a459)

**Reply by sebastianmusial** [reply]
@rmcsharry sorry for a late answer. Did you receive any errors in the console? It looks like a temporary issue on the openai side. Could you confirm that it already already works?
